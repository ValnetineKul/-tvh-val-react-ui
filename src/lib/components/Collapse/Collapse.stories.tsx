import React, { useState } from 'react';
import { makeStyles } from '../../../themes/core';
import Button from '../Buttons/Button';
import Divider from '../Divider';
import { AngleDown, AngleUp } from '../Icon/icons/functional';
import Icon from '../Icon';
import Typography from '../Typography';
import Collapse from './Collapse';
import { createMeta } from '../../story-utils';

const useStyles = makeStyles()((theme) => ({
  root: {
    paddingLeft: theme.layoutSpacing['spacing/400'],
    paddingBlock: theme.layoutSpacing['spacing/400'],
  },
  row: {
    display: 'flex',
    justifyContent: 'space-between',
  },
}));

const Template = (args: React.ComponentProps<typeof Collapse>) => {
  const [isOpen, setIsOpen] = useState<boolean>(false);
  const { classes } = useStyles();

  return (
    <>
      <Button
        variant="link"
        onClick={() => setIsOpen(!isOpen)}
        label="Open"
        endIcon={<Icon icon={isOpen ? AngleUp : AngleDown} />}
      />
      <Collapse {...args} open={isOpen}>
        <div className={classes.root}>
          <div className={classes.row}>
            <Typography weight="emphasis">Total products</Typography>
            <Typography>$ 7,000.30</Typography>
          </div>

          <div className={classes.row}>
            <Typography weight="emphasis">Est. taxes</Typography>
            <Typography>$ 3.30</Typography>
          </div>

          <div className={classes.row}>
            <Typography weight="emphasis">Est. fees</Typography>
            <Typography>$ 32.30</Typography>
          </div>
        </div>
      </Collapse>
      <Divider />
    </>
  );
};

export const Primitive = Template.bind({});
Primitive.storyName = 'Collapse';

export default createMeta({
  component: Collapse,
  title: 'Components/Collapse/Collapse',
  argTypes: {
    open: {
      control: {
        disable: true,
      },
    },
  },
});
