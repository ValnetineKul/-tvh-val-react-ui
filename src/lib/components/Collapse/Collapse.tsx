import React from 'react';
import type { FC } from 'react';
import { default as MuiCollapse } from '@mui/material/Collapse';

export interface CollapseProps {
  open: boolean;
  className?: string;
  children: React.ReactNode;
  collapsedSize?: string | number;
}

const Collapse: FC<CollapseProps> = ({ collapsedSize, open = false, className, children }) => {
  return (
    <MuiCollapse in={open} collapsedSize={collapsedSize} className={className}>
      {children}
    </MuiCollapse>
  );
};

export default Collapse;
