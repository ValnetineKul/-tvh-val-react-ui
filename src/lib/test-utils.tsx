import userEvent from '@testing-library/user-event';
import { fireEvent, render } from '@testing-library/react';
import { renderHook } from '@testing-library/react-hooks';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { theme } from '../themes/tvh';
import { createTheme, ThemeProvider, SurfaceProvider } from '../themes/core';

export const muiTheme = createTheme(theme);

const AllTheProviders: React.FC = ({ children }) => {
  return (
    <MemoryRouter>
      <ThemeProvider theme={muiTheme}>
        <SurfaceProvider color="100">{children}</SurfaceProvider>
      </ThemeProvider>
    </MemoryRouter>
  );
};

const customRender = (component: React.ReactElement, options?: Parameters<typeof render>[1]) =>
  render(component, { wrapper: AllTheProviders, ...options });

const customRenderHook: typeof renderHook = (hook, options) =>
  renderHook(hook, { wrapper: AllTheProviders, ...options });

export const initResponsiveTest = (size: number | 'Mobile' | 'Tablet' | 'Desktop' | 'LgDesktop' | 'XlDesktop') => {
  let width: number;
  if (size === 'Mobile') width = muiTheme.breakpoints.values.sm - 1;
  else if (size === 'Tablet') width = muiTheme.breakpoints.values.sm;
  else if (size === 'Desktop') width = muiTheme.breakpoints.values.md;
  else if (size === 'LgDesktop') width = muiTheme.breakpoints.values.lg;
  else if (size === 'XlDesktop') width = muiTheme.breakpoints.values.xl;
  else width = size;

  Object.defineProperty(window, 'innerWidth', {
    writable: true,
    configurable: true,
    value: width,
  });

  window.matchMedia = jest.fn().mockImplementation((query) => {
    const queryWidthInterval: string[] = query.match(/\d+(\.\d+)?/g);

    let matches;
    if (queryWidthInterval.length === 2) {
      matches = width >= +queryWidthInterval[0] && width < +queryWidthInterval[1];
    } else {
      const queryWidth = +queryWidthInterval[0];
      const isDown = /max/.test(query);

      matches = isDown ? width <= queryWidth : width >= queryWidth;
    }

    return {
      matches,
      media: query,
      onchange: null,
      addListener: jest.fn(),
      removeListener: jest.fn(),
    };
  });
};

const customUserEvent = Object.assign(userEvent, {
  mouseDown: fireEvent.mouseDown,
  mouseUp: fireEvent.mouseUp,
  mouseMove: fireEvent.mouseMove,
  scroll: fireEvent.scroll,
  error: fireEvent.error,
});

export { screen, within, waitFor, act, waitForElementToBeRemoved } from '@testing-library/react';
// override render method
export { customRender as render, customRenderHook as renderHook, customUserEvent as userEvent };
