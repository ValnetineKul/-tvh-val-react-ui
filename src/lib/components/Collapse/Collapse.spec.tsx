import React from 'react';
import { render } from '../../test-utils';
import Collapse from './Collapse';

describe('Container', () => {
  it('Should render correctly', () => {
    const { container } = render(
      <Collapse open>
        <div>children</div>
      </Collapse>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
