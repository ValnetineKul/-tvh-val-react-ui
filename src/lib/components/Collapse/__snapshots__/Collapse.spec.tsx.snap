// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container Should render correctly 1`] = `
<div
  class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered mui-pwcg7p-MuiCollapse-root"
  style="min-height: 0px;"
>
  <div
    class="MuiCollapse-wrapper MuiCollapse-vertical mui-smkl36-MuiCollapse-wrapper"
  >
    <div
      class="MuiCollapse-wrapperInner MuiCollapse-vertical mui-9l5vo-MuiCollapse-wrapperInner"
    >
      <div>
        children
      </div>
    </div>
  </div>
</div>
`;
